atlp-rwanda/ecommerce-app-legends-fe
# Ch setup ci/cd pipeline

## What does this PR do?
This PR sets up a CI/CD pipeline with Github Actions for our React project. It includes Codeclimate test coverage and badges for maintainability, code coverage, and Github Actions CI.

## Description of Task to be completed?
The task was to create a CI/CD pipeline with Github Actions and integrate Codeclimate test coverage for our React project. We also wanted to display badges for maintainability, code coverage, and Github Actions CI in the project's README file.

## How should this be manually tested?
To test this PR, you can do the following:
1. Clone the repository on your local machine.
2. Checkout the branch associated with this PR.
3. Run the `npm install` command to install the required dependencies.
4. Run the `npm test` command to run the tests and generate the coverage report.
5. Check the Codeclimate dashboard for the updated test coverage report.
6. Check the project's README file for the updated badges.

## Any background context you want to provide?
We wanted to automate the deployment process of our React project and ensure that our code quality and test coverage metrics are always up to standard. This PR accomplishes that goal by setting up a CI/CD pipeline with Github Actions and integrating Codeclimate test coverage.

## What are the relevant pivotal tracker/Trello stories?
This PR is not associated with any pivotal tracker/Trello stories.
